test/postman.test.ts

```diff
@@ -0,0 +1,120 @@
+      signer,
+    );
+    mockStarknetMessaging = await MockStarknetMessaging.deploy();
+    await mockStarknetMessaging.deployed();
+
+    L1L2Example = await ethers.getContractFactory('L1L2Example', signer);
+    l1l2Example = await L1L2Example.deploy(mockStarknetMessaging.address);
+    await l1l2Example.deployed();
+  });
+
+  it('should deploy the messaging contract', async () => {
+    const {
+      address: deployedTo,
+      l1_provider: L1Provider,
+    } = await starknet.devnet.loadL1MessagingContract(networkUrl);
+
+    assert.exists(deployedTo);
+    assert(L1Provider === networkUrl, `L1 provider is ${L1Provider}, should be ${networkUrl}`);
+  });
+
+  it('should load the already deployed contract if the address is provided', async () => {
+
+    const {
+      address: loadedFrom,
+    } = await starknet.devnet.loadL1MessagingContract(
+      networkUrl,
+      mockStarknetMessaging.address,
+    );
+
+    assert(mockStarknetMessaging.address === loadedFrom, `Loaded from address is ${loadedFrom}, should be ${mockStarknetMessaging.address}`);
+  });
+
+  it('should exchange messages between L1 and L2', async () => {
+    // Load mock messaging contract
+    await starknet.devnet.loadL1MessagingContract(
+      networkUrl,
+      mockStarknetMessaging.address,
+    );
+
+    await l2contract.invoke('increase_balance', {
+      user,
+      amount: 100,
+    });
+    await l2contract.invoke('withdraw', {
+      user,
+      amount: 10,
+      L1_CONTRACT_ADDRESS: BigInt(l1l2Example.address),
+    });
+    let getBalanceResponse = await l2contract.call('get_balance', {
+      user,
+    });
+
+    assert(getBalanceResponse.balance === 90n, `L2 balance should be 90, but got ${getBalanceResponse.balance}`);
+
+    const {
+      n_consumed_l2_to_l1_messages,
+    } = await starknet.devnet.flush();
+
+    assert(n_consumed_l2_to_l1_messages === 1, `${n_consumed_l2_to_l1_messages} messages were consumed, should be 1.`);
+
+    let userBalanceL1: BigNumber = await l1l2Example.userBalances(user);
+
+    assert(userBalanceL1.eq(0), `Initial L1 should be 0, but got ${userBalanceL1}`);
+
+    await l1l2Example.withdraw(l2contract.address, user, 10);
+    userBalanceL1 = await l1l2Example.userBalances(user);
+
+    assert(userBalanceL1.eq(10), `After the withdraw of 10 L1 balance should be 10, but got ${userBalanceL1}`);
+
+    await l1l2Example.deposit(l2contract.address, user, 2);
+
+    userBalanceL1 = await l1l2Example.userBalances(user);
+
+    assert(userBalanceL1.eq(8), `After the deposit of 2 L1 balance should be 8, but got ${userBalanceL1}`);
+
+    getBalanceResponse = await l2contract.call('get_balance', {
+      user,
+    });
+
+    assert(getBalanceResponse.balance === 90n, `L2 balance should stay 90 before the flush, but got ${getBalanceResponse.balance}`);
+
+    // Flush gets called by the devnet when the L2 contract method is invoked
+
+    getBalanceResponse = await l2contract.call('get_balance', {
```

Comment:
Isn't it better to keep flush explicit? That is how postman works in python. Flush pushes messages in both directions?

Reply:
Yes, `flush` pushes messages in both direction. As for keeping the `flush` explicit, I think @dribeiro-ShardLabs  could answer that one

Reply:
Ideally we would find a way of making the flush process automatic for both ways, since that's the way it works on alpha, but we don't have a clear way to check for new messages coming from L1 into postman. But since we do know when an L2 method is invoked we can flush it at the end of the transaction automatically in devnet, which we feel like it's a better UX than having the user explicitly have to flush for both directions.

Reply:
StarkNet messaging contract emits LogMessageToL2 on L1->L2 messages. You can listen and flush. Then it will be fully automatic.

Reply:
@maciejka Thanks for the idea, we'll move on with the explicit `flush` functionality, but we plan to do the improvements in the future. Hopefully the current implementation will be enough for you to start testing the L1 <> L2 communication!

Reply:
It seems to be enough. Thanks a lot. When are you going to release it?